.golangci.yml

```diff
@@ -0,0 +1,106 @@
+run:
+  skip-files:
+    - pkg/dns/dns.go
+
+linters-settings:
+  dupl:
+    threshold: 100
+  funlen:
+    lines: 100
+    statements: 50
+  goconst:
+    min-len: 2
+    min-occurrences: 3
+  gocritic:
+    enabled-tags:
+      - diagnostic
+      - experimental
+      - opinionated
+      - performance
+      - style
+    disabled-checks:
+      - dupImport # https://github.com/go-critic/go-critic/issues/845
+      - ifElseChain
+      - octalLiteral
+      - whyNoLint
+      - wrapperFunc
+  gocyclo:
+    min-complexity: 15
+  gomnd:
+    # don't include the "operation" and "assign"
+    checks:
+      - argument
+      - case
+      - condition
+      - return
+    ignored-numbers:
+      - '0'
+      - '1'
+      - '2'
+      - '3'
+    ignored-functions:
+      - strings.SplitN
+
+  lll:
+    line-length: 140
+  misspell:
+    locale: US
+  nolintlint:
+    allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
+    allow-unused: false # report any unused nolint directives
+    require-explanation: false # don't require an explanation for nolint directives
```

Comment:
Why not?

Reply:
I personally think it would make a mess in the codebase. One option is to move the explanations to the golanci.yml file.

Reply:
I agree that `gochecknoinits` doesn't need explanations (in which case, [`allow-no-explanation`](https://golangci-lint.run/usage/linters#nolintlint) would be useful); I'm not sure yet whether we'll end up with too many lints with obvious reasons… maybe we can come back to this later.